search/index.go

```diff
@@ -0,0 +1,244 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	api "github.com/tigrisdata/tigris-client-go/api/server/v1"
+	"github.com/tigrisdata/tigris-client-go/driver"
+	"github.com/tigrisdata/tigris-client-go/filter"
+	"github.com/tigrisdata/tigris-client-go/schema"
+)
+
+// Collation allows you to specify string comparison rules. Default is case-sensitive.
+type Collation struct {
+	// Case is case-sensitive by default, to perform case-insensitive query override it and set
+	// this to 'ci'.
+	Case string
+}
+
+// Index provides an interface for documents manipulation.
+// Such as Insert, Update, Delete, Read.
+type Index[T schema.Model] struct {
+	Name         string
+	SearchClient driver.SearchClient
+}
+
+func getSearch(_ context.Context, crud driver.SearchClient) driver.SearchClient {
+	return crud
+}
+
+// DeleteIndex drops the index.
+func (c *Index[T]) DeleteIndex(ctx context.Context) error {
```

Comment:
Can we have DeleteIndex on Search object?

Reply:
Ok. I was also thinking it's a bit unnatural to destroy itself.